### PR TITLE
fix: instrumentation entries should not contain user labels

### PR DESCRIPTION
**Pull request by daniel-sanche**
Previously, instrumentation logs would inherit labels and other log metadata from the parent log. This was causing issues when the user expected logs with certain labels to conform to a set structure, which the instrumentation log would break

Instead, instrumentation logs should be static, containing only a severity field

### CODE REVIEW

File: google/cloud/logging_v2/_instrumentation.py [hunk 1]
```diff
@@ -67,8 +67,12 @@ def _create_diagnostic_entry(name=_PYTHON_LIBRARY_NAME, version=_LIBRARY_VERSION
             _INSTRUMENTATION_SOURCE_KEY: [_get_instrumentation_source(name, version)]
         }
     }
-    kw["severity"] = "INFO"
-    entry = StructEntry(payload=payload, **kw)
+    active_kws = {"severity": "INFO"}
+    # only keep the log_name and resource from the parent log
+    for key in ["log_name", "resource"]:
```
Comment (daniel-sanche) [inline google/cloud/logging_v2/_instrumentation.py:72]: @losalex Are there any other fields that should be included? Do the other languages keep the resource field?

File: google/cloud/logging_v2/_instrumentation.py [hunk 2]
```diff
@@ -67,8 +67,12 @@ def _create_diagnostic_entry(name=_PYTHON_LIBRARY_NAME, version=_LIBRARY_VERSION
             _INSTRUMENTATION_SOURCE_KEY: [_get_instrumentation_source(name, version)]
         }
     }
-    kw["severity"] = "INFO"
-    entry = StructEntry(payload=payload, **kw)
+    active_kws = {"severity": "INFO"}
```
Comment (losalex) [inline google/cloud/logging_v2/_instrumentation.py:70]: I believe the severity should be avoided, thus will remain default - that is what we have in other languages

File: google/cloud/logging_v2/_instrumentation.py (hunk 1 above)
Comment (losalex) [reply google/cloud/logging_v2/_instrumentation.py:72]: Those are mandatory ones and those are the only ones which should be included.

File: tests/unit/test__instrumentation.py [hunk 3]
```diff
@@ -63,3 +63,16 @@ def test_truncate_long_values(self):
 
         self.assertEqual(expected_name, self._get_diagonstic_value(entry, "name"))
         self.assertEqual(expected_version, self._get_diagonstic_value(entry, "version"))
+
+    def test_drop_labels(self):
```
Comment (losalex) [inline tests/unit/test__instrumentation.py:67]: Would it be possible to add a test which checks that only required fields are present?

**Review by losalex — APPROVED** [review]
LGTM. Please see my comments regarding severity and extra test checking only required fields are present

File: tests/unit/test__instrumentation.py (hunk 3 above)
Comment (daniel-sanche) [reply tests/unit/test__instrumentation.py:67]: yup, added an assert

File: google/cloud/logging_v2/_instrumentation.py (hunk 2 above)
Comment (daniel-sanche) [reply google/cloud/logging_v2/_instrumentation.py]: done

**Pull request merged**